src/pyseekdb/client/client_base.py

```
@@ -1324,6 +1520,72 @@ def _collection_fork(self, collection: Collection, forked_name: str) -> None:
+                        f"but value at index {i} is not a string: {type(text).__name__}"
+                    )
+                source_texts.append(text)
+            else:
+                return [None] * num_items
+
+        # Generate sparse embeddings
+        logger.debug(f"Generating sparse embeddings for {len(source_texts)} items")
+        try:
+            sparse_vectors = sparse_ef(source_texts)
+        except Exception as e:
+            raise ValueError(f"Failed to generate sparse embeddings: {e}") from e
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)